### PR TITLE
Remove spree preferences

### PR DESCRIPTION
**Pull request by mbj**
Depends on:

* [x] #644 
* [x] #643

### CODE REVIEW

File: core/app/models/spree/gateway.rb [hunk 1]
```diff
@@ -26,7 +23,7 @@ def provider
     end
 
     def options
-      self.preferences.inject({}){ |memo, (key, value)| memo[key.to_sym] = value; memo }
+      {}
```
Comment (mbj) [inline core/app/models/spree/gateway.rb]: For our main apps gateways we'll have to provide these in a custom manner.

But it'll force us into a superior approach where we can avoid having to read important credentials for the DB and can put the credentials under version control via our provisioning mechanism.

**Comment by mbj** [comment]
@dkubb This is ready for a code only review.

File: core/app/models/spree/gateway.rb (hunk 1 above)
Comment (dkubb) [inline core/app/models/spree/gateway.rb]: Should the options be frozen to discourage mutation?

Comment (mbj) [inline core/app/models/spree/gateway.rb]: Lets introduce the `Spree::EMPTY_HASH` good call.

**Comment by mbj** [comment]
@dkubb This is ready for another pass, we should merge in only after #637 because of the coordination the main app side of this change will require (reading the gateway options from outside the DB).

**Comment by dkubb** [comment]
> This is ready for another pass, we should merge in only after #637

@mbj I don't think there's any overlap between that PR and this one. I think if you're able to get the mrh-spree side of this ready before, then it could get merged sooner.

**Comment by mbj** [comment]
> @mbj I don't think there's any overlap between that PR and this one. I think if you're able to get the mrh-spree side of this ready before, then it could get merged sooner.

I do think the main app side of this will take longer than the main app side of #637, to not cause an unneded dependency I do want to hold this one back.

**Comment by mbj** [comment]
@dkubb This is ready for review / merge.

**Pull request merged**